PWGLF/TableProducer/v0qaanalysis.cxx

```diff
@@ -125,82 +132,157 @@ struct v0qaanalysis {
-      float ctauK0s = v0.distovertotmom(collision.posX(), collision.posY(), collision.posZ()) * RecoDecay::getMassPDG(310);
+      // Generated particles
+      const auto particlesInCollision = mcParticles.sliceByCached(aod::mcparticle::mcCollisionId, collision.mcCollision().globalIndex(), cache1);
 
-      if (v0.v0radius() > v0radius &&
-          v0.v0cosPA(collision.posX(), collision.posY(), collision.posZ()) > v0cospa &&
-          TMath::Abs(v0.posTrack_as<DauTracksMC>().eta()) < etadau &&
-          TMath::Abs(v0.negTrack_as<DauTracksMC>().eta()) < etadau // &&
-      ) {
+      for (auto& mcParticle : particlesInCollision) {
+        if (std::abs(mcParticle.y()) > rapidity) {
+          continue;
+        }
+        if (!mcParticle.isPhysicalPrimary()) {
+          continue;
+        }
+        if (mcParticle.pdgCode() == 310)
+          registry.fill(HIST("Generated_MCRecoColl_K0Short"), mcParticle.pt()); // K0s
+        if (mcParticle.pdgCode() == 3122)
+          registry.fill(HIST("Generated_MCRecoColl_Lambda"), mcParticle.pt()); // Lambda
+        if (mcParticle.pdgCode() == -3122)
+          registry.fill(HIST("Generated_MCRecoColl_AntiLambda"), mcParticle.pt()); // AntiLambda
+      }
+    }
 
-        float cent = 0.;
+    for (const auto& mccollision : mcCollisions) {
 
-        // Fill table
-        myv0s(v0.globalIndex(), v0.pt(), v0.yLambda(), v0.yK0Short(),
-              v0.mLambda(), v0.mAntiLambda(), v0.mK0Short(),
-              v0.v0radius(), v0.v0cosPA(collision.posX(), collision.posY(), collision.posZ()),
-              v0.dcapostopv(), v0.dcanegtopv(), v0.dcaV0daughters(),
-              v0.posTrack_as<DauTracksMC>().eta(), v0.negTrack_as<DauTracksMC>().eta(),
-              v0.posTrack_as<DauTracksMC>().phi(), v0.negTrack_as<DauTracksMC>().phi(),
-              posITSNhits, negITSNhits, ctauLambda, ctauAntiLambda, ctauK0s,
-              v0.negTrack_as<DauTracksMC>().tpcNSigmaPr(), v0.posTrack_as<DauTracksMC>().tpcNSigmaPr(),
-              v0.negTrack_as<DauTracksMC>().tpcNSigmaPi(), v0.posTrack_as<DauTracksMC>().tpcNSigmaPi(),
-              v0.negTrack_as<DauTracksMC>().tofNSigmaPr(), v0.posTrack_as<DauTracksMC>().tofNSigmaPr(),
-              v0.negTrack_as<DauTracksMC>().tofNSigmaPi(), v0.posTrack_as<DauTracksMC>().tofNSigmaPi(),
-              v0.posTrack_as<DauTracksMC>().hasTOF(), v0.negTrack_as<DauTracksMC>().hasTOF(), lPDG, isprimary,
-              cent, cent);
+      if (TMath::Abs(mccollision.posZ()) > cutzvertex) {
+        continue;
       }
-    }
 
-    for (auto& mcparticle : mcParticles) {
+      bool isFT0A = false;
+      bool isFT0C = false;
+
+      const auto particlesInMCCollision = mcParticles.sliceByCached(aod::mcparticle::mcCollisionId, mccollision.globalIndex(), cache2);
+
+      for (auto& mcParticle : particlesInMCCollision) {
+
+        if (!mcParticle.isPhysicalPrimary()) {
+          continue;
+        }
+
+        if (std::abs(mcParticle.pdgCode() == 211)) {
```

Comment:
You have a logical bug here, it should be `std::abs(mcParticle.pdgCode()) == 211`

Reply:
Hi @aalkin thanks a lot for spotting this! I fixed it in this PR: https://github.com/AliceO2Group/O2Physics/pull/2919